src/BrazeKit-dev.js

```diff
@@ -709,6 +719,108 @@ var constructor = function () {
         }
     }
 
+    function prepareInitialConsent(user) {
+        var userConsentState = getUserConsentState(user);
+
+        var currentConsentPayload = generateBrazeConsentStatePayload(
+            userConsentState
+        );
+
+        latestUserBrazeConsentString = JSON.stringify(currentConsentPayload);
+
+        if (latestUserBrazeConsentString.length) {
+            setConsentOnBraze(currentConsentPayload);
+        }
+    }
+
+    function setConsentOnBraze(currentConsentPayload) {
+        for (var key in currentConsentPayload) {
+            braze
+                .getUser()
+                .setCustomUserAttribute(key, currentConsentPayload[key]);
+        }
+    }
+
+    function maybeSetConsentBeforeEventLogged(event) {
+        if (latestUserBrazeConsentString && parsedConsentMappings.length) {
+            var eventConsentState = getEventConsentState(event.ConsentState);
+
+            if (!isEmpty(eventConsentState)) {
```

Comment:
if we do end up doing the suggested change mentioned above this can be simplified and to one if statement with the one above i,e
`if (
        latestUserBrazeConsentString &&
        parsedConsentMappings.length &&
        !this.isEmpty(eventConsentState)
    )`
And we can remove `var eventConsentState = getEventConsentState(event.ConsentState);`
just following Alex's tip on not having unnecessary nested if statements

Reply:
My only pushback on this suggestion is that if you have too many parameters in a conditional, it becomes less readable and harder to understand. In this case though, I don't mind as long as we don't add more than 3.

Reply:
@mmustafa-tse @alexs-mparticle - The way I looked at it, what I did is a minor optimization. Only if there is a `latestConsentString` and there are no `parsedConsentMappings` should we even try to look at `event.ConsentState`.  So if they dont exist, we'd be running an addiitonal function every time.  Even though it's O(1) either way, I figured if we can skip it, why not?